File: 136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int> single;
        for(int i=0;i<nums.size();i++){
          if(!single[nums[i]]){
              single[nums[i]]=1;
          }
          else{
              single[nums[i]]++;
          }
        }
        for(int i=0;i<nums.size();i++){
            if(single[nums[i]]==1){
                return nums[i];
            }
        }
        return -1;
    }
};

File: 136-single-number/single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let hash = {};
    let n = nums.length;
    for(let i=0;i<n;i++){
        if(!hash[nums[i]]){
            hash[nums[i]] =1;
        }
        else{
            hash[nums[i]]++;
        }
    }
    for(let i=0;i<n;i++){
        if(hash[nums[i]]==1){
            return nums[i];
        }
    }
};